### PR TITLE
Add drag and drop reordering of initiative count

### LINKED ISSUE #6: Ability to manually reorder the initiative order outside of rolled order
The initiative order should have some sort of "click and drag" reordering such that things can be reordered at the DM's discretion. 

The main use case for this is if there are ties in the rolled initiative order. It is nice to be able to let players work out who will go first in these situations.

### PR DESCRIPTION
**Pull request by mspoehr**
Closes #6

### CODE REVIEW

File: src/pages/RunEncounter/DisplayInitiative.tsx [hunk 1]
```diff
@@ -56,6 +65,30 @@ const DisplayInitiative: React.FunctionComponent = () => {
...
+    draggableStyle: NotDraggingStyle | DraggingStyle | undefined
+  ): React.CSSProperties => ({
+    // some basic styles to make the items look a bit nicer
+    userSelect: "none",
+    marginBottom: "1rem",
+
+    // styles we need to apply on draggables
+    ...draggableStyle,
+
+    // change background color if dragging
```
Comment (cymbre1) [inline src/pages/RunEncounter/DisplayInitiative.tsx:88]: Apparently this doesn't even make sense

File: src/pages/RunEncounter/DisplayInitiative.tsx [hunk 2]
```diff
@@ -56,6 +65,30 @@ const DisplayInitiative: React.FunctionComponent = () => {
...
+
+  const getItemStyle = (
+    isDragging: boolean,
+    draggableStyle: NotDraggingStyle | DraggingStyle | undefined
+  ): React.CSSProperties => ({
+    // some basic styles to make the items look a bit nicer
+    userSelect: "none",
+    marginBottom: "1rem",
+
+    // styles we need to apply on draggables
```
Comment (cymbre1) [inline src/pages/RunEncounter/DisplayInitiative.tsx:85]: I feel like the name is descriptive enough that you don't need this comment

File: src/redux/initiative-tracker/initiativeTrackerSlice.ts [hunk 3]
```diff
@@ -27,12 +53,11 @@ export const initiativeTrackerSlice = createSlice({
...
-
+    addCreature: (state, action: PayloadAction<InitiativeCreatureExternal>) => {
       const currentTurnUuid = state.creatures[state.currentTurn]?.uuid;
 
-      state.creatures.push(action.payload);
+      const creature: InitiativeCreature = { order: 0, uuid: uuidv4(), ...action.payload };
+      state.creatures.push(creature);
       sortInitiativeCreatures(state.creatures);
 
       // If combat has started, preserve the turn of the creature whose turn it currently is
```
Comment (cymbre1) [inline src/redux/initiative-tracker/initiativeTrackerSlice.ts:63]: Why is this comment here. Is this a todo?

**Review by cymbre1 — CHANGES_REQUESTED** [review]

File: src/redux/initiative-tracker/initiativeTrackerSlice.ts (hunk 3 above)
Comment (mspoehr) [reply src/redux/initiative-tracker/initiativeTrackerSlice.ts:63]: clarifies the logic below it

File: src/pages/RunEncounter/DisplayInitiative.tsx (hunk 1 above)
Comment (mspoehr) [reply src/pages/RunEncounter/DisplayInitiative.tsx]: Addressed

File: src/pages/RunEncounter/DisplayInitiative.tsx (hunk 2 above)
Comment (mspoehr) [reply src/pages/RunEncounter/DisplayInitiative.tsx]: Addressed

**Review by cymbre1 — APPROVED** [review]

**Pull request merged**